src/container/container_iv.c

```diff
@@ -181,14 +198,10 @@ cont_iv_ent_destroy(d_sg_list_t *sgl)
 		daos_handle_t *root_hdl = sgl->sg_iovs[0].iov_buf;
 		int rc;
 
-		while (!dbtree_is_empty(*root_hdl)) {
-			rc = dbtree_iterate(*root_hdl, DAOS_INTENT_PUNCH, false,
-					    delete_iter_cb, NULL);
-			if (rc < 0) {
-				D_ERROR("dbtree iterate fails %d\n", rc);
-				return rc;
-			}
-		}
+		rc = delete_whole_tree(*root_hdl);
+		if (rc)
+			return rc;
+
 		dbtree_destroy(*root_hdl, NULL);
```

Comment:
The tree will be destroyed anyway, above delete_whole_tree() isn't necessary.

Reply:
right